File: c language/question 45.c
// 45. Given a sentence, return the longest word in it.
// Input: "I love programming in JavaScript"
// Output: "programming"
// #include <stdio.h>
// int main(){

// int i,j,k;
// char input[] = "I love jenilsavaliab JavaScript j ";
// int itt = 0;
// char arr[4];


// for(i=0;i<strlen(input);i++){

// 	if(input[i] == ' '){
// 	arr[itt] = i;
// 	itt = itt + 1;
// 	printf("%d\n",i);
// 	}
	
// }


// int max_diff = abs(arr[1] - arr[0]);
// int index1 = 0, index2 = 1;
// int arr_len = sizeof(arr)/sizeof(arr[0]);


// for(j=0;j<arr_len;j++){
	
// 	int diff = abs(arr[j+1]-arr[j]);
//     printf("%d\n",arr[j+1]-arr[j]);

//     if(diff > max_diff){
// 	  max_diff = diff;
// 	  index1=j;
// 	  index2=j+1;
// 	}

// }

// printf("%d\n",index1);
// printf("%d\n",index2);
// for(k=arr[index1];k<arr[index2];k++){
// 	printf("%c",input[k]);
// }


// }

File: Java script/question 45.js
// 45. Given a sentence, return the longest word in it.
// Input: "I love programming in JavaScript"
// Output: "programming"
// var input ="I love programming in JavaScript"
// var len=input.length
// var cnt=0
// var cntmax=0
// var output=""
// for(i=0;i<len;i++){
//     if(input[i]!=" "){
//         cnt++
//     }
//     else{
//         cnt=0
//     }
//     output=output+input[i]
//     if(cnt>cntmax){
//         cntmax=cnt
//        }
//     }
// console.log(output)


// var input ="I love programming in JavaScript"
// var len=input.length
// var arr=[]

// var cnt=0
// var cntmax=0
// var output=""
// for(i=0;i<len;i++){
//     if(input[i]==" "){
//         arr.push(i)
//     }
// }
// var start=0
// var end=0
// maxdiff=Math.abs(arr[0]-arr[1])
// for(j=0;j<arr.length;j++){
//     var diff=Math.abs(arr[j+1]-arr[j])
//     if(diff>maxdiff){
//         maxdiff=diff
//         start=j
//         end=j+1
//     }
// }
// for(k=arr[start];k<arr[end];k++){
//     console.log(input[k])
// }